src/proxy_handler.cpp

```diff
@@ -0,0 +1,100 @@
+#include <string>
+#include <istream>
+#include <ostream>
+#include <boost/asio.hpp>
+#include "proxy_handler.h"
+
+using boost::asio::ip::tcp;
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
+         const NginxConfig& config) {
+	uri_prefix_ = uri_prefix;
+	RequestHandler::ParseConfig(config);
+	host = ProxyHandler::getHost();
+	portno = ProxyHandler::getPort();
+
+	return OK;
+}
+
+RequestHandler::Status ProxyHandler::HandleRequest(const Request& request,
+		Response* response) {
+    //This function is inspired by www.boost.org/doc/libs/1_49_0/doc/html/boost_asio/example/http/client/sync_client.cpp
+    std::string uri;
+    if (request.uri() == uri_prefix_)
+    uri = "/" ;
```

Comment:
Please enclose these in curly braces as the professors have discussed in class multiple times ! :)